src/core/highlight.js

```diff
@@ -45,32 +45,39 @@ export async function run(conf) {
         console.error("Timed-out waiting for highlight:", element);
         done();
       }, 4000);
-      const msg = {
-        action: "highlight",
-        code: element.textContent,
-        id: `highlight:${i}`,
-        languages: getLanguageHint(element.classList),
-      };
-      worker.addEventListener("message", function listener(ev) {
-        const {
-          data: { id, language, value },
-        } = ev;
-        if (id !== msg.id) {
-          return; // not for us!
-        }
-        element.innerHTML = value;
-        if (element.localName === "pre") {
-          element.classList.add("hljs");
-        }
-        if (language) {
-          element.classList.add(language);
-        }
-        clearTimeout(timeoutId);
-        worker.removeEventListener("message", listener);
-        done();
-      });
-      element.setAttribute("aria-busy", "true");
-      worker.postMessage(msg);
+      const elementsToHighlight = element.querySelectorAll("code").length > 0 ? element.querySelectorAll("code") : [element];
+      for(let each of elementsToHighlight) {
```

Comment:
Also here we have a problem we cant just loop since we have a `eventListener` ! The contents of the first `<code>` element gets overwritten into the content of the `<code>` below it due to the delay of response from the worker!

Reply:
Right, I think you should do the following... add this function outside this function: 

```JS
function makeListener(id){
  return function listener(ev) {
     // current listener code
  } 
}
```

Reply:
I think the problem is the `<code>`s inside a single `<pre>` share the same `id` for highlight request.

Reply:
ah yes, the id should increment per postMessage.

Reply:
Done! Fixed that issue, pushed the code, I'll just add the tests ...